pkg/cmd/server/kubernetes/master_config.go

```diff
@@ -166,67 +179,109 @@ func BuildDefaultAPIServer(options configapi.MasterConfig) (*apiserveroptions.Se
 		autoscaling.SchemeGroupVersion,
 	)
 
-	storageGroupsToEncodingVersion, err := server.GenericServerRunOptions.StorageGroupsToEncodingVersion()
+	storageGroupsToEncodingVersion, err := server.StorageSerialization.StorageGroupsToEncodingVersion()
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
 	// use the stock storage config based on args, but override bits from our config where appropriate
-	etcdConfig := server.GenericServerRunOptions.StorageConfig
-	etcdConfig.Prefix = options.EtcdStorageConfig.KubernetesStoragePrefix
-	etcdConfig.ServerList = options.EtcdClientInfo.URLs
-	etcdConfig.KeyFile = options.EtcdClientInfo.ClientCert.KeyFile
-	etcdConfig.CertFile = options.EtcdClientInfo.ClientCert.CertFile
-	etcdConfig.CAFile = options.EtcdClientInfo.CA
-
-	storageFactory, err := genericapiserver.BuildDefaultStorageFactory(
+	etcdConfig := server.Etcd.StorageConfig
+	etcdConfig.Prefix = masterConfig.EtcdStorageConfig.KubernetesStoragePrefix
+	etcdConfig.ServerList = masterConfig.EtcdClientInfo.URLs
+	etcdConfig.KeyFile = masterConfig.EtcdClientInfo.ClientCert.KeyFile
+	etcdConfig.CertFile = masterConfig.EtcdClientInfo.ClientCert.CertFile
+	etcdConfig.CAFile = masterConfig.EtcdClientInfo.CA
+
+	storageFactory, err := kubeapiserver.NewStorageFactory(
```

Comment:
Does this mean you want to drop everything `server.APIEnablement.RuntimeConfig` does?

Then we end up with this: https://github.com/sttts/origin/pull/70/commits/f57955d6ad141822a82fbd4eb6807bed5208c7bd